distributed/shuffle/_rechunk.py

```diff
@@ -156,32 +155,25 @@ class ShardID(NamedTuple):
     shard_index: NIndex
 
 
-def rechunk_slicing(
-    old: ChunkedAxes, new: ChunkedAxes
-) -> dict[NIndex, list[tuple[ShardID, NSlice]]]:
-    """Calculate how to slice the old chunks to create the new chunks
+SplitChunk: TypeAlias = list[tuple[int, int, slice]]
+SplitAxis: TypeAlias = list[SplitChunk]
+SplitAxes: TypeAlias = list[SplitAxis]
 
-    Returns
-    -------
-        Mapping of each old chunk to a list of tuples defining where each slice
-        of the old chunk belongs. Each tuple consists of the index
-        of the new chunk, the index of the slice within the composition of slices
-        creating the new chunk, and the slice to be applied to the old chunk.
-    """
-    from dask.array.rechunk import intersect_chunks
-
-    ndim = len(old)
-    intersections = intersect_chunks(old, new)
-    new_indices = product(*(range(len(c)) for c in new))
-
-    slicing = defaultdict(list)
 
-    for new_index, new_chunk in zip(new_indices, intersections):
-        sub_shape = [len({slice[dim][0] for slice in new_chunk}) for dim in range(ndim)]
+def split_axes(old: ChunkedAxes, new: ChunkedAxes) -> SplitAxes:
+    from dask.array.rechunk import old_to_new
 
-        shard_indices = product(*(range(dim) for dim in sub_shape))
+    _old_to_new = old_to_new(old, new)
 
-        for shard_index, sliced_chunk in zip(shard_indices, new_chunk):
-            old_index, nslice = zip(*sliced_chunk)
-            slicing[old_index].append((ShardID(new_index, shard_index), nslice))
-    return slicing
+    axes = []
+    for axis_id, new_axis in enumerate(_old_to_new):
+        old_axis: SplitAxis = [[] for _ in old[axis_id]]
+        for new_chunk_id, new_chunk in enumerate(new_axis):
```

Comment:
OK, so this is O(n^2) which makes sense because it's a transpose-like operation.